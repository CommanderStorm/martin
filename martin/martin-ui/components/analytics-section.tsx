import { Image, Layers, Palette, Type } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Skeleton } from "@/components/ui/skeleton";
import type { AnalyticsData } from "@/lib/types";

interface AnalyticsSectionProps {
  analytics?: AnalyticsData;
  isLoading?: boolean;
  error?: Error | null;
  onRetry?: () => void;
  isRetrying?: boolean;
}

const CARD_CONFIG = [
  {
    icon: Layers,
    key: "tiles",
    title: "Tiles",
  },
  {
    icon: Palette,
    key: "styles",
    title: "Styles",
  },
  {
    icon: Type,
    key: "fonts",
    title: "Fonts",
  },
  {
    icon: Image,
    key: "sprites",
    title: "Sprites",
  },
] as const;

export function AnalyticsSection({
  analytics,
  isLoading,
  error = null,
  onRetry,
  isRetrying = false,
}: AnalyticsSectionProps) {
  if (error) {
    return (
      <div className="mb-8">
        <ErrorState
          description="Unable to fetch server metrics and usage data"
          error={error}
          isRetrying={isRetrying}
          onRetry={onRetry}
          showDetails={true}
          title="Failed to Load Analytics"
          variant="server"
        />
      </div>
    );
  }

  return (
    <div className="space-y-6 mb-8">
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
        {CARD_CONFIG.map(({ key, title, icon: Icon }) => {
          const data = analytics?.[key as keyof AnalyticsData];
          return (
            <Card key={key}>
              <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
                <CardTitle className="text-sm font-medium">{title}</CardTitle>
                <Icon className="h-4 w-4 text-primary" />
              </CardHeader>
              <CardContent>
                <div className="text-2xl font-bold flex flex-row gap-1">
                  {isLoading || !data ? (
                    <Skeleton className="h-6 w-8 flex flex-row" />
                  ) : (
                    Math.round(data.averageRequestDurationMs)
                  )}{" ms"}
                </div>
                <p className="text-xs text-muted-foreground flex flex-row gap-1">
                  {isLoading || !data ? (
                    <Skeleton className="h-3 w-12" />
                  ) : (
                    data.requestCount.toLocaleString()
                  )}{" requests"}
                </p>
              </CardContent>
            </Card>
          );
        })}
      </div>
    </div>
  );
}
