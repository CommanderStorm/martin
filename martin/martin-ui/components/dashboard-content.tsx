"use client";

import { useRouter, useSearchParams } from "next/navigation";
import { type ErrorInfo, useCallback, useEffect } from "react";
import { FontCatalog } from "@/components/catalogs/font";
import { SpriteCatalog } from "@/components/catalogs/sprite";
import { StylesCatalog } from "@/components/catalogs/styles";
import { TilesCatalog } from "@/components/catalogs/tiles";
import { ErrorBoundary } from "@/components/error/error-boundary";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { Toaster } from "@/components/ui/toaster";
import { useAsyncOperation } from "@/hooks/use-async-operation";
import { useToast } from "@/hooks/use-toast";
import { buildMartinUrl } from "@/lib/api";
import { getMartinMockCatalog } from "@/lib/mockResponses";
import type { CatalogSchema } from "@/lib/types";

const fetchCatalog = async (): Promise<CatalogSchema> => {
  if (process.env.NEXT_PUBLIC_MARTIN_ENABLE_MOCK_API === "true") {
    return getMartinMockCatalog();
  }

  const res = await fetch(buildMartinUrl("/catalog"));
  if (!res.ok) {
    throw new Error(`Failed to fetch catalog: ${res.statusText}`);
  }
  return res.json();
};

export function DashboardContent() {
  const { toast } = useToast();
  const router = useRouter();
  const searchParams = useSearchParams();

  const searchQuery = searchParams.get("search") || "";
  // Update search param in URL, preserving tab and other params
  const setSearchQuery = (query: string) => {
    const params = new URLSearchParams(Array.from(searchParams.entries()));
    if (query) {
      params.set("search", query);
    } else {
      params.delete("search");
    }
    router.replace(`?${params.toString()}`, { scroll: false });
  };

  const handleTabChange = (value: string) => {
    const params = new URLSearchParams(Array.from(searchParams.entries()));
    params.set("tab", value);
    router.replace(`?${params.toString()}`, { scroll: false });
  };

  const handleCatalogError = useCallback((error: Error) => {
    console.error("Catalog fetch failed:", error);
  }, []);

  // Catalog operation - unified data fetching
  const catalogOperation = useAsyncOperation<CatalogSchema>(fetchCatalog, {
    onError: handleCatalogError,
    showErrorToast: false,
  });

  // Load catalog data
  useEffect(() => {
    catalogOperation.execute();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [catalogOperation.execute]);

  return (
    <ErrorBoundary
      onError={(error: Error, errorInfo: ErrorInfo) => {
        console.error("Application error:", error, errorInfo);
        toast({
          description: "An unexpected error occurred. The page will reload automatically.",
          title: "Application Error",
          variant: "destructive",
        });

        // Auto-reload after 3 seconds
        setTimeout(() => {
          window.location.reload();
        }, 3000);
      }}
    >
      <Tabs
        className="space-y-6"
        onValueChange={handleTabChange}
        value={searchParams.get("tab") || "tiles"}
      >
        <TabsList className="grid w-full grid-cols-4">
          <TabsTrigger value="tiles">Data Catalog</TabsTrigger>
          <TabsTrigger value="styles">Styles Catalog</TabsTrigger>
          <TabsTrigger value="fonts">Font Catalog</TabsTrigger>
          <TabsTrigger value="sprites">Sprite Catalog</TabsTrigger>
        </TabsList>

        <TabsContent value="tiles">
          <TilesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            tileSources={catalogOperation.data?.tiles}
          />
        </TabsContent>

        <TabsContent value="styles">
          <StylesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            styles={catalogOperation.data?.styles}
          />
        </TabsContent>

        <TabsContent value="fonts">
          <FontCatalog
            error={catalogOperation.error}
            fonts={catalogOperation.data?.fonts}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
          />
        </TabsContent>

        <TabsContent value="sprites">
          <SpriteCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            spriteCollections={catalogOperation.data?.sprites}
          />
        </TabsContent>
      </Tabs>

      <Toaster />
    </ErrorBoundary>
  );
}
