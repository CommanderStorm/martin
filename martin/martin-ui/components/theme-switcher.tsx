"use client";

import { Moon, Sun, SunMoon } from "lucide-react";
import { useTheme } from "next-themes";
import * as React from "react";
import { Button } from "./ui/button";
import {
  Tooltip,
  TooltipContent,
  TooltipTrigger,
} from "./ui/tooltip";

const themeOrder = ["light", "dark", "system"] as const;

export function ThemeSwitcher() {
	const { theme, setTheme } = useTheme();

	const getNextTheme = () => {
		const idx = themeOrder.indexOf((theme as typeof themeOrder[number]) || "system");
		return themeOrder[(idx + 1) % themeOrder.length];
	};

	const getIcon = () => {
		switch (theme) {
			case "light":
				return <Sun className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
			case "dark":
				return <Moon className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
			default:
				return <SunMoon className="h-[1.2rem] w-[1.2rem]" aria-hidden="true" />;
		}
	};

	const getLabel = () => {
		switch (theme) {
			case "light":
				return "Switch to dark theme";
			case "dark":
				return "Switch to system theme";
			default:
				return "Switch to light theme";
		}
	};

	return (
		<Tooltip>
			<TooltipTrigger asChild>
				<Button
					variant="outline"
					size="icon"
					onClick={() => setTheme(getNextTheme())}
					aria-label={getLabel()}
				>
					{getIcon()}
				</Button>
			</TooltipTrigger>
			<TooltipContent>
				<p>{getLabel()}</p>
			</TooltipContent>
		</Tooltip>
	);
}
