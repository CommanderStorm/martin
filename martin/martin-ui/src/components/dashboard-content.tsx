import { type ErrorInfo, useCallback, useEffect, useState } from "react";
import { FontCatalog } from "@/components/catalogs/font";
import { SpriteCatalog } from "@/components/catalogs/sprite";
import { StylesCatalog } from "@/components/catalogs/styles";
import { TilesCatalog } from "@/components/catalogs/tiles";
import { ErrorBoundary } from "@/components/error/error-boundary";
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { Toaster } from "@/components/ui/toaster";
import { useAsyncOperation } from "@/hooks/use-async-operation";
import { useToast } from "@/hooks/use-toast";
import { buildMartinUrl } from "@/lib/api";
import type { CatalogSchema } from "@/lib/types";

const fetchCatalog = async (): Promise<CatalogSchema> => {
  const res = await fetch(buildMartinUrl("/catalog"));
  if (!res.ok) {
    throw new Error(`Failed to fetch catalog: ${res.statusText}`);
  }
  return res.json();
};

export function DashboardContent() {
  const { toast } = useToast();
  const [activeTab, setActiveTab] = useState("tiles");
  const [searchQuery, setSearchQuery] = useState("");

  const handleTabChange = (value: string) => {
    setActiveTab(value);
  };

  const handleCatalogError = useCallback((error: Error) => {
    console.error("Catalog fetch failed:", error);
  }, []);

  // Catalog operation - unified data fetching
  const catalogOperation = useAsyncOperation<CatalogSchema>(fetchCatalog, {
    onError: handleCatalogError,
    showErrorToast: false,
  });

  // Load catalog data
  useEffect(() => {
    catalogOperation.execute();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [catalogOperation.execute]);

  return (
    <ErrorBoundary
      onError={(error: Error, errorInfo: ErrorInfo) => {
        console.error("Application error:", error, errorInfo);
        toast({
          description: "An unexpected error occurred. The page will reload automatically.",
          title: "Application Error",
          variant: "destructive",
        });

        // Auto-reload after 3 seconds
        setTimeout(() => {
          window.location.reload();
        }, 3000);
      }}
    >
      <Tabs className="space-y-6" onValueChange={handleTabChange} value={activeTab}>
        <TabsList className="grid w-full grid-cols-4">
          <TabsTrigger value="tiles">Data Catalog</TabsTrigger>
          <TabsTrigger value="styles">Styles Catalog</TabsTrigger>
          <TabsTrigger value="fonts">Font Catalog</TabsTrigger>
          <TabsTrigger value="sprites">Sprite Catalog</TabsTrigger>
        </TabsList>

        <TabsContent value="tiles">
          <TilesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            tileSources={catalogOperation.data?.tiles}
          />
        </TabsContent>

        <TabsContent value="styles">
          <StylesCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            styles={catalogOperation.data?.styles}
          />
        </TabsContent>

        <TabsContent value="fonts">
          <FontCatalog
            error={catalogOperation.error}
            fonts={catalogOperation.data?.fonts}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
          />
        </TabsContent>

        <TabsContent value="sprites">
          <SpriteCatalog
            error={catalogOperation.error}
            isLoading={catalogOperation.isLoading}
            onSearchChangeAction={setSearchQuery}
            searchQuery={searchQuery}
            spriteCollections={catalogOperation.data?.sprites}
          />
        </TabsContent>
      </Tabs>

      <Toaster />
    </ErrorBoundary>
  );
}
