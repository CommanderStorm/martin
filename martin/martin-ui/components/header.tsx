"use client";

import { BookOpen, Info } from "lucide-react";
import Image from "next/image";
import Link from "next/link";
import { useTheme } from "next-themes";
import { Badge } from "@/components/ui/badge";
import { HoverCard, HoverCardContent, HoverCardTrigger } from "@/components/ui/hover-card";
import { ThemeSwitcher } from "./theme-switcher";
import { Skeleton } from "./ui/skeleton";

export function Header() {
  const { theme } = useTheme();
  return (
    <header className="border-b bg-navbar backdrop-blur-md backdrop-brightness-90 sticky top-0 z-50">
      <div className="container mx-auto px-6">
        <div className="flex h-20 items-center justify-between">
          <div className="flex items-center space-x-4">
            <Link className="items-center flex" href="/">
              {theme ?
              <Image
                alt="Martin Logo"
                className="-mt-5 -rotate-6"
                height={48}
                priority
                src={`/logo_martin-${theme}.svg`}
                title="Martin Tile Server"
                width={96}
              />:<Skeleton className="h-12 w-24" />}
              <h1 className="text-3xl -ms-2 font-bold leading-relaxed text-foreground select-none">
                MARTIN
              </h1>
            </Link>
            <Badge asChild className="hover:bg-purple-700 hidden md:block" variant="default">
              <Link
                className="p-1"
                href={`https://github.com/maplibre/martin/releases/tag/${process.env.NEXT_PUBLIC_VERSION}`}
              >
                {process.env.NEXT_PUBLIC_VERSION}
              </Link>
            </Badge>
          </div>
          <div className="flex items-center space-x-6">
            <HoverCard>
              <HoverCardTrigger asChild>
                <Link
                  className="flex items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
                  href="https://maplibre.org/martin/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <BookOpen size={18} />
                  <span>Documentation</span>
                </Link>
              </HoverCardTrigger>
              <HoverCardContent className="w-80">
                <div className="space-y-2">
                  <h4 className="text-sm font-semibold">Martin Documentation</h4>
                  <p className="text-sm">
                    Access comprehensive guides and documentation for the Martin tile server.
                  </p>
                </div>
              </HoverCardContent>
            </HoverCard>
            <HoverCard>
              <HoverCardTrigger asChild>
                <Link
                  className="md:flex hidden items-center gap-2 text-foreground hover:bg-accent hover:text-accent-foreground px-3 py-2 rounded-md transition-all"
                  href="https://maplibre.org"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <Info size={18} />
                  <span>About us</span>
                </Link>
              </HoverCardTrigger>
              <HoverCardContent className="w-80">
                <div className="space-y-2">
                  <h4 className="text-sm font-semibold">About MapLibre</h4>
                  <p className="text-sm">
                    Learn about MapLibre,
                    <br />
                    the open-source collective behind this project.
                  </p>
                </div>
              </HoverCardContent>
            </HoverCard>
            <ThemeSwitcher />
          </div>
        </div>
      </div>
    </header>
  );
}
