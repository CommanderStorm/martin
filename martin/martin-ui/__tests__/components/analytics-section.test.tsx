import type React from "react";
import { AnalyticsSection } from "@/components/analytics-section";
import type { AnalyticsData } from "@/lib/types";
import { fireEvent, render, screen } from "../test-utils";

// Mock icons and UI components to avoid unnecessary complexity in snapshots
jest.mock("lucide-react", () => ({
  Image: () => <div data-testid="icon-image" />,
  Layers: () => <div data-testid="icon-layers" />,
  Palette: () => <div data-testid="icon-palette" />,
  Type: () => <div data-testid="icon-type" />,
}));
jest.mock("@/components/ui/card", () => ({
  Card: ({ children }: { children: React.ReactNode }) => <div data-testid="card">{children}</div>,
  CardContent: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="card-content">{children}</div>
  ),
  CardHeader: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="card-header">{children}</div>
  ),
  CardTitle: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="card-title">{children}</div>
  ),
}));
jest.mock("@/components/ui/skeleton", () => ({
  Skeleton: ({ className }: { className?: string }) => (
    <div className={className} data-testid="skeleton" />
  ),
}));
jest.mock("@/components/error/error-state", () => ({
  ErrorState: (props: any) => (
    <div data-testid="error-state">
      {props.title} - {props.description} - {props.error?.message || props.error}
      <button disabled={props.isRetrying} onClick={props.onRetry}>
        Retry
      </button>
    </div>
  ),
}));

const analytics: AnalyticsData = {
  sprites: {
    averageRequestDurationMs: 10.5,
    requestCount: 1000,
  },
  tiles: {
    averageRequestDurationMs: 25.2,
    requestCount: 50000,
  },
  fonts: {
    averageRequestDurationMs: 5.1,
    requestCount: 500,
  },
  styles: {
    averageRequestDurationMs: 2.3,
    requestCount: 200,
  },
};

describe("AnalyticsSection", () => {
  it("renders loading state (skeletons)", () => {
    const { container } = render(<AnalyticsSection isLoading />);
    expect(container).toMatchSnapshot();
    expect(screen.getAllByTestId("skeleton").length).toBeGreaterThan(0);
  });

  it("renders analytics data", () => {
    const { container } = render(<AnalyticsSection analytics={analytics} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText("11 ms")).toBeInTheDocument();
    expect(screen.getByText("1,000 requests")).toBeInTheDocument();
    expect(screen.getByText("25 ms")).toBeInTheDocument();
    expect(screen.getByText("50,000 requests")).toBeInTheDocument();
    expect(screen.getByText("5 ms")).toBeInTheDocument();
    expect(screen.getByText("500 requests")).toBeInTheDocument();
    expect(screen.getByText("2 ms")).toBeInTheDocument();
    expect(screen.getByText("200 requests")).toBeInTheDocument();
  });

  it("renders error state and calls onRetry", () => {
    const onRetry = jest.fn();
    const error = new Error("Test error");
    render(<AnalyticsSection error={error} isRetrying={false} onRetry={onRetry} />);
    expect(screen.getByTestId("error-state")).toHaveTextContent("Test error");
    fireEvent.click(screen.getByText("Retry"));
    expect(onRetry).toHaveBeenCalled();
  });

  it("renders retrying state (button disabled)", () => {
    const onRetry = jest.fn();
    const error = new Error("Retry error");
    render(<AnalyticsSection error={error} isRetrying={true} onRetry={onRetry} />);
    expect(screen.getByText("Retry")).toBeDisabled();
  });
});
