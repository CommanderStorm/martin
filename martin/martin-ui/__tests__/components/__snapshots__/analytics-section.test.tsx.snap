// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AnalyticsSection renders analytics data 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,l,o,d,u,h)=&gt;{let m=document.documentElement,w=["light","dark"];function p(r){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&d?o.map(f=&gt;d[f]||f):o;k?(m.classList.remove(...S),m.classList.add(d&&d[r]?d[r]:r)):m.setAttribute(y,r)}),C(r)}function C(r){h&&w.includes(r)&&(m.style.colorScheme=r)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(l)p(l);else try{let r=localStorage.getItem(s)||n,y=u&&r==="system"?a():r;p(y)}catch(r){}})("class","theme","light",null,["light","dark"],null,true,true)
  </script>
  <div
    class="space-y-6 mb-8"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6"
    >
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Sprites
          </div>
          <div
            data-testid="icon-image"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="text-2xl font-bold"
          >
            11
             ms
          </div>
          <p
            class="text-xs text-muted-foreground"
          >
            1,000
             requests
          </p>
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Tiles
          </div>
          <div
            data-testid="icon-layers"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="text-2xl font-bold"
          >
            25
             ms
          </div>
          <p
            class="text-xs text-muted-foreground"
          >
            50,000
             requests
          </p>
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Fonts
          </div>
          <div
            data-testid="icon-type"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="text-2xl font-bold"
          >
            5
             ms
          </div>
          <p
            class="text-xs text-muted-foreground"
          >
            500
             requests
          </p>
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Styles
          </div>
          <div
            data-testid="icon-palette"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="text-2xl font-bold"
          >
            2
             ms
          </div>
          <p
            class="text-xs text-muted-foreground"
          >
            200
             requests
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AnalyticsSection renders loading state (skeletons) 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,l,o,d,u,h)=&gt;{let m=document.documentElement,w=["light","dark"];function p(r){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&d?o.map(f=&gt;d[f]||f):o;k?(m.classList.remove(...S),m.classList.add(d&&d[r]?d[r]:r)):m.setAttribute(y,r)}),C(r)}function C(r){h&&w.includes(r)&&(m.style.colorScheme=r)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(l)p(l);else try{let r=localStorage.getItem(s)||n,y=u&&r==="system"?a():r;p(y)}catch(r){}})("class","theme","light",null,["light","dark"],null,true,true)
  </script>
  <div
    class="space-y-6 mb-8"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6"
    >
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Sprites
          </div>
          <div
            data-testid="icon-image"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="h-8 w-24 mb-2"
            data-testid="skeleton"
          />
          <div
            class="h-3 w-32"
            data-testid="skeleton"
          />
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Tiles
          </div>
          <div
            data-testid="icon-layers"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="h-8 w-24 mb-2"
            data-testid="skeleton"
          />
          <div
            class="h-3 w-32"
            data-testid="skeleton"
          />
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Fonts
          </div>
          <div
            data-testid="icon-type"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="h-8 w-24 mb-2"
            data-testid="skeleton"
          />
          <div
            class="h-3 w-32"
            data-testid="skeleton"
          />
        </div>
      </div>
      <div
        data-testid="card"
      >
        <div
          data-testid="card-header"
        >
          <div
            data-testid="card-title"
          >
            Styles
          </div>
          <div
            data-testid="icon-palette"
          />
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="h-8 w-24 mb-2"
            data-testid="skeleton"
          />
          <div
            class="h-3 w-32"
            data-testid="skeleton"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
